src/Uno.Extensions.Toolkit.UI/UIElementExtensions.cs

```diff
@@ -4,4 +4,29 @@ public static class UIElementExtensions
 {
 	public static IThemeService GetThemeService(this UIElement element, ILogger? logger = default) =>
 		new ThemeService(() => element.XamlRoot, new Dispatcher(element), logger);
+
+	public static Task<IThemeService> GetInitializedThemeServiceAsync(this UIElement element, ILogger? logger = default) =>
+		element.InitializeElementThemeServiceAsync(element.GetThemeService(logger));
+
+	internal static async Task<IThemeService> InitializeElementThemeServiceAsync(this UIElement element, IThemeService themeService)
+	{
+		if (element is null) return themeService;
+
+		if (element.XamlRoot is { } xamlroot)
+		{
+			await themeService.InitializeAsync();
+		}
+		else
+		{
+			if (element is FrameworkElement fe)
+			{
+				fe.Loaded += async (_, _) =>
+				{
+					await themeService.InitializeAsync();
+				};
+			}
```

Comment:
```suggestion
			if (element is FrameworkElement fe)
			{
			    async void OnLoaded(object sender, EventArgs args)
			    {
			        fe.Loaded -= OnLoaded;
			        await themeService.InitializeAsync();
			    }
				fe.Loaded += OnLoaded;
			}
```
Let's be sure we don't create memory leaks and unhook from events. Not to mention we only need to initialize once.